### DISCUSSION TITLE
How to configure auto closing the outline window when an item selected?

### DISCUSSION THREAD

**Discussion by Rom888** [neoclide/coc.nvim#4925] [Q&A] [answered]
How to configure auto closing the outline window when an item selected?

I didn't find anything here:
h coc-outline
h coc-config-tree
h coc-config-outline

**Comment by fannheyward** [comment, marked as answer]
~No configuration to auto closing the outline for now.~

You can use `outline.autoHide` to enable auto closing.

You can use `:call CocAction('hideOutline')` to close it.

**Comment by Rom888** [comment]
Thank you, is it possible to add this feature?

**Reply by fannheyward** [reply]
https://github.com/neoclide/coc.nvim/pull/4926, add `outline.autoHide`.

**Reply by Rom888** [reply]
It works now, you are genius, thank you!

**Reply by Rom888** [reply]
Unfortunately there is an issue with
"outline.autoHide": true,

It works, but when I switch to another tab and then return back I see that outline window is active again.

gvim version 9.1.65
@fannheyward

**Reply by fannheyward** [reply]
> I switch to another tab and then return back

Can't reproduce. My steps:

1. `nvim x`
2. `:CocOutline`, opens outline window for file x
3. now, `<cr>` to select one, the outline window is closed
4. `:tabnew y`
5. `gt`, back to x, the outline window is closed.

Your steps please.

**Reply by Rom888** [reply]
The same steps:
1. gvim 9.1.65 undo Ubuntu 23.04
2. :CocConfig
3. :CocOutline, opens outline window for file coc-settings.json
4. now, <cr> to select one, the outline window is closed
5. :tabnew y
6. gt, back to x, the outline window is open again.

I tried to run gvim without .vimrc, but with the same result.
Also I tried these steps in Vim in a terminal, but with the same result.

**Reply by Rom888** [reply]
If I just close an outline window using ctrl+w+c I don't see this problem.
@fannheyward

**Reply by Rom888** [reply]
Also I tried to remove all other plugins from ~/.vim/pack/vendor/start but without success.

**Reply by fannheyward** [reply]
Reproduced with vim, not nvim

**Reply by eliasdorneles** [reply]
Not sure if it's related, but I am using [the ToggleOutline() function as explained in the docs](https://github.com/neoclide/coc.nvim/blob/29774cf47307a39a0d5c667068c377cef2253eea/doc/coc.txt#L1387-L1398), and the outline window kept reopening for me when switching back from another tab.

I've switched from `coc#window#close(winid)` to `win_execute(winid, 'close')` and the problem is fixed.

The full snippet from my config:

```
" map CTRL_T to toggle outline view
nnoremap <silent><nowait> <C-t>  :call ToggleOutline()<CR>
function! ToggleOutline() abort
    let winid = coc#window#find('cocViewId', 'OUTLINE')
    if winid == -1
      call CocActionAsync('showOutline')
    else
      " note: using win_execute instead of coc#window#close because the latter
      " reopens the outline window when switching back to the buffer
      call win_execute(winid, 'close')
    endif
endfunction
```
